Fix #3199 by honoring alpha

englercj opened the pull request:
Proposed to fix #3199, as an alternative to #3941.

Using this test code:

```js
'use strict';

const app = new PIXI.Application({
    backgroundColor: 0x006600,
    width: 512,
    height: 512,
    view: document.getElementById('view'),
});

const graphic = new PIXI.Graphics()
    .beginFill(0xff0000)
    .drawCircle(256, 256, 200);

graphic.filters = [new PIXI.filters.NoiseFilter()];

app.stage.addChild(graphic);
```

Produces this output:

![image](https://cloud.githubusercontent.com/assets/944497/25076095/cdcc3264-22d2-11e7-9560-ebb54a72339a.png)

bigtimebuddy commented:
I tested this out and adjusting the `noise` property doesn't seem to do anything. I believe this is suppose to be a float from 0 to 1 of the intensity of the noise. For instance:

![noise](https://cloud.githubusercontent.com/assets/864393/25089552/2e429830-234c-11e7-8b6f-2ea4ddd68716.png)

englercj commented:
I used it as a seed, not as intensity, so you could do this:

```
app.ticker.add(() => {
    graphic.filters[0].noise = Math.random();
});
```

Which will result in:

![2017-04-17_8-01-51](https://cloud.githubusercontent.com/assets/944497/25092835/484778d4-2344-11e7-874a-d84c57235367.gif)

bigtimebuddy commented:
It's just that this is a breaking change. That `noise` setter currently is the amount of noise, not the randomness/seed. I would think that should be a separate property.

englercj commented:
Alrighty, I'll try to fix it up Soon.

englercj commented:
Updated, now adjusting the `noise` property looks like:

![adjust-noise](https://cloud.githubusercontent.com/assets/944497/25094225/f51904d2-234a-11e7-81ad-ff28114e42af.gif)

And I added a `seed` property, which when adjusted gives you some actual noise:

![adjust-seed-and-noise](https://cloud.githubusercontent.com/assets/944497/25094227/f95f279c-234a-11e7-929b-e8f8b7377ade.gif)

bigtimebuddy commented:
Wow, so awesome!

bigtimebuddy commented on src/filters/noise/NoiseFilter.js:
```diff
@@ -29,21 +29,37 @@ export default class NoiseFilter extends core.Filter
         );
 
         this.noise = 0.5;
```
One very small thing that would be nice is to allow a single constructor argument for the noise value with a default for `0.5`. 

```js
// Current, non-default value
const filter = new PIXI.filters.NoiseFilter();
filter.noise = 0.1;
displayObject.filters = [ filter ];

// With constructor arg
displayObject.filters = [ new PIXI.filters.NoiseFilter(0.1) ];
```

bigtimebuddy replied:
We could do this as another PR too, but thought I'd toss it in.

englercj commented:
Added ctor args for `noise` and `seed`.

bigtimebuddy commented:
@englercj https://travis-ci.org/pixijs/pixi.js/builds/222841155#L345-L346

englercj commented:
Damn I suck, I'll fix it when I get home :P

englercj commented:
Fixed the jsdoc issue, and the branch is updating now. Then should be good to go!
